pkg/storage/stores/shipper/util/util.go

```diff
@@ -235,3 +241,100 @@ func QueryKey(q chunk.IndexQuery) string {
 
 	return ret
 }
+
+func IsCompressedFile(filename string) bool {
+	return strings.HasSuffix(filename, ".gz")
+}
+
+func DoConcurrentWork(ctx context.Context, maxConcurrency, workCount int, logger log.Logger, do func(workNum int) error) error {
```

Comment:
You can use https://github.com/grafana/dskit/blob/main/concurrency/runner.go#L65 instead.

jobs []interface{} = [0,1,2,3,....,workCount]int